plugins/inputs/http_response/http_response.go

```diff
@@ -137,6 +146,25 @@ func (h *HTTPResponse) HTTPGather() (map[string]interface{}, error) {
 	}
 	fields["response_time"] = time.Since(start).Seconds()
 	fields["http_response_code"] = resp.StatusCode
+
+	// Check the response for a regex match
+	if h.ResponseStringMatch != "" {
+		regex, err := regexp.Compile(h.ResponseStringMatch)
+		if err != nil {
+			log.Printf("E! Failed to compile regular expression %s : %s", h.ResponseStringMatch, err)
+			fields["response_string_match"] = 0
+		} else {
+
+			bodyBytes, _ := ioutil.ReadAll(resp.Body)
+			bodyString := string(bodyBytes)
```

Comment:
do not convert the response body to a string, you can use regex.Match below (instead of regex.MatchString)